src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskCompleteByIdControllerTest.java

```diff
@@ -147,137 +115,6 @@ public void should_return_a_403_if_task_was_not_previously_assigned() {
-            HttpStatus.NO_CONTENT
-        );
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            taskId,
-            caseworkerCredentials.getHeaders()
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.NO_CONTENT.value());
-
-        assertions.taskVariableWasUpdated(taskVariables.getProcessInstanceId(), "taskState", "completed");
-        assertions.taskStateWasUpdatedInDatabase(taskId, "completed", caseworkerCredentials.getHeaders());
-
-        common.cleanUpTask(taskId);
-
-    }
-
-    @Test
-    public void should_return_a_401_when_the_user_did_not_have_any_roles() {
-
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "processApplication");
-        taskId = taskVariables.getTaskId();
-        initiateTask(taskVariables, Jurisdiction.WA);
-
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            taskId,
-            caseworkerCredentials.getHeaders()
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.UNAUTHORIZED.value())
-            .contentType(APPLICATION_JSON_VALUE)
-            .body("timestamp", lessThanOrEqualTo(ZonedDateTime.now().plusSeconds(60)
-                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
-            .body("error", equalTo(HttpStatus.UNAUTHORIZED.getReasonPhrase()))
-            .body("status", equalTo(HttpStatus.UNAUTHORIZED.value()))
-            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
-
-        common.cleanUpTask(taskId);
-
-    }
-
-    @Test
-    public void should_return_a_204_and_retrieve_a_task_by_id_jurisdiction_location_and_region_match() {
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "processApplication");
-        taskId = taskVariables.getTaskId();
-        initiateTask(taskVariables, Jurisdiction.WA);
-        common.setupOrganisationalRoleAssignmentWithCustomAttributes(
-            caseworkerCredentials.getHeaders(),
-            Map.of(
-                "primaryLocation", "765324",
-                "jurisdiction", "WA"
-            )
-        );
-
-        given.iClaimATaskWithIdAndAuthorization(
-            taskId,
-            caseworkerCredentials.getHeaders(),
-            HttpStatus.NO_CONTENT
-        );
-
-        Response result = restApiActions.post(
-            ENDPOINT_BEING_TESTED,
-            taskId,
-            caseworkerCredentials.getHeaders()
-        );
-
-        result.then().assertThat()
-            .statusCode(HttpStatus.NO_CONTENT.value());
-        assertions.taskStateWasUpdatedInDatabase(taskId, "completed", caseworkerCredentials.getHeaders());
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void should_return_a_403_when_the_user_did_not_have_sufficient_permission_region_did_not_match() {
```

Comment:
Is there an IT tests? I.e. where region did not match?

Reply:
We can discuss a common test that can test role assignment verification base on attributes. I have added one here
https://github.com/hmcts/wa-task-management-api/pull/812/files#diff-9f243eae8e68fcb2ca4be82bae3797ed1f0ef6085d8f651d946b4cfc9a175d3fR448

But I think this should be tested separately from task actions

Reply:
Re-introduce. Address in new ticket